pkg/ir/spec.go

```diff
@@ -174,3 +192,25 @@ func (s *ConfigDefs) RemoteFromIP(ip IP) RemoteType {
 type Reader interface {
 	ReadSpec(filename string, defs *ConfigDefs) (*Spec, error)
 }
+
+func cidrsAsIPs(cidrSegments map[string]map[string][]string, segmentName string) []IP {
+	retVal := make([]IP, 0)
+	for cidr := range cidrSegments[segmentName] {
+		retVal = append(retVal, IPFromString(cidr))
+	}
+	return retVal
+}
+
+func (s *ConfigDefs) SubnetsContainedInCidr(cidr ipblocks.IPBlock) ([]string, error) {
+	var containedSubnets []string
+	for subnet, ip := range s.Subnets {
+		subnetIPBlock, err := ipblocks.NewIPBlockFromCidrOrAddress(ip.String())
+		if err != nil {
+			return nil, err
+		}
+		if subnetIPBlock.ContainedIn(&cidr) {
+			containedSubnets = append(containedSubnets, subnet)
+		}
+	}
+	return containedSubnets, nil
```

Comment:
```suggestion
	}
	sort.Strings(containedSubnets)
	return containedSubnets, nil
```
I believe this should fix the nondeterminism issue.